app/controllers/stars_controller.rb

```diff
@@ -0,0 +1,40 @@
+class StarsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[create]
+  before_action :set_star, only: %i[destroy]
+
+  def create
+    star = Star.new(post: @post, user: current_user)
+
+    if star.save
+      flash[:notice] = 'Star created successfully!'
+    else
+      flash[:alert] = star.errors.full_messages.to_sentence if star.errors
+    end
+
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy
+    if @star.destroy
+      flash[:notice] = 'Star destroyed successfully!'
+    else
+      flash[:alert] = star.errors.full_messages.to_sentence if star.errors
+    end
+
+    redirect_back(fallback_location: root_path)
+  end
```

Comment:
The intent of `redirect_back` here is it saves all of the params. So if you star a post on page 6, you won't be redirected back to page 1 and lose your spot in your story hunt 🕵🏽